src/Components/Web/src/Forms/EditForm.cs

```diff
@@ -180,12 +174,29 @@ void RenderFormContents(RenderTreeBuilder builder, ModelBindingContext? bindingC
             builder.OpenComponent<CascadingValue<EditContext>>(5);
             builder.AddComponentParameter(6, "IsFixed", true);
             builder.AddComponentParameter(7, "Value", _editContext);
-            builder.AddComponentParameter(8, "ChildContent", ChildContent?.Invoke(_editContext));
+            if (bindingContext != null && !OperatingSystem.IsBrowser())
+            {
+                builder.AddComponentParameter(8, "ChildContent", RenderWithBindingValidator());
+            }
+            else
+            {
+                builder.AddComponentParameter(8, "ChildContent", ChildContent?.Invoke(_editContext));
+            }
             builder.CloseComponent();
             builder.CloseElement();
         }
     }
 
+    private RenderFragment? RenderWithBindingValidator()
+    {
+        return builder =>
+        {
+            builder.OpenComponent<ModelBindingContextValidator>(1);
+            builder.CloseComponent();
+            builder.AddContent(2, ChildContent!, EditContext);
+        };
+    }
```

Comment:
Minor detail, but I think this doesn't have to create a new closure on each invocation. It could be:

```cs
private readonly RenderTreeBuilder _renderWithBindingValidator = (builder) => {
            builder.OpenComponent<ModelBindingContextValidator>(1);
            builder.CloseComponent();
            builder.AddContent(2, ChildContent!, EditContext);
};
```

... or some other form like that. Then line 179 can use it directly as a delegate instance without having to invoke a method that creates a delegate.

Reply:
done